### PR TITLE
feat: Alerting add alert builder veo state

### PR DESCRIPTION
**Pull request by ebb-tide**
partially closes: #13923

### CODE REVIEW

File: ui/src/alerting/components/NewCheckEO.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    </Overlay>
+  )
+}
+
+const mstp = (state: AppState): StateProps => {
+  const {
+    checks: {
+      current: {check, status},
+    },
+  } = state
```
Comment (121watts) [inline ui/src/alerting/components/NewCheckEO.tsx]: what happens if any of these are undefined?

File: ui/src/dashboards/components/EditVEO.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  onSetActiveTimeMachine,
+  getViewForTimeMachine,
+  executeQueries,
+  loadingState,
+  onSaveView,
+  onSetName,
+  params,
+  router,
+  views,
+  view,
```
Comment (121watts) [inline ui/src/dashboards/components/EditVEO.tsx]: This 🎄 is upside down but ill take it!

File: ui/src/dashboards/components/EditVEO.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+    executeQueries()
+  }, [get(view, 'id', null)])
+
+  const handleClose = () => {
+    const {orgID, dashboardID} = params
+    router.push(`/orgs/${orgID}/dashboards/${dashboardID}`)
+  }
+
+  const handleSave = () => {
+    onSaveView(params.dashboardID)
```
Comment (121watts) [inline ui/src/dashboards/components/EditVEO.tsx]: what if the save errors?  do we still close?

File: ui/src/dashboards/components/EditVEO.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      </div>
+    </Overlay>
+  )
+}
+
+const mstp = (state: AppState, {params: {cellID}}): StateProps => {
+  const {activeTimeMachineID} = state.timeMachines
+  const {view} = getActiveTimeMachine(state)
+
+  const viewMatchesRoute = get(view, 'id', null) == cellID
```
Comment (121watts) [inline ui/src/dashboards/components/EditVEO.tsx]: `===`

File: ui/src/dataExplorer/components/DataExplorer.tsx [hunk 5]
```diff
@@ -11,7 +11,7 @@ import AssetLimitAlert from 'src/cloud/components/AssetLimitAlert'
 import {setActiveTimeMachine} from 'src/timeMachine/actions'
 
 // Utils
-import {DE_TIME_MACHINE_ID} from 'src/timeMachine/constants'
+import {TimeMachineEnum} from 'src/timeMachine/constants'
```
Comment (121watts) [inline ui/src/dataExplorer/components/DataExplorer.tsx]: vnice!

File: ui/src/timeMachine/constants/index.ts [hunk 6]
```diff
@@ -1,2 +1,5 @@
-export const DE_TIME_MACHINE_ID = 'de'
-export const VEO_TIME_MACHINE_ID = 'veo'
+export enum TimeMachineEnum {
+  DE = 'de',
+  VEO = 'veo',
+  Alerting = 'alerting',
```
Comment (121watts) [inline ui/src/timeMachine/constants/index.ts]: `ALERT` maybe?

**Comment by kelwang** [comment]
Lgtm

File: ui/src/alerting/components/NewCheckEO.tsx (hunk 1 above)
Comment (ebb-tide) [reply ui/src/alerting/components/NewCheckEO.tsx]: I think this component is protected agains either being undefined.

**Pull request merged**